src/crimson/os/seastore/journal/circular_bounded_journal.cc

```diff
@@ -262,111 +366,50 @@ Journal::replay_ret CircularBoundedJournal::replay(
     header = head;
     DEBUG("header : {}", header);
     initialized = true;
-    written_to.segment_seq = NULL_SEG_SEQ;
-    auto tail = get_dirty_tail() <= get_alloc_tail() ?
-      get_dirty_tail() : get_alloc_tail();
-    set_written_to(tail);
     return seastar::do_with(
-      bool(false),
-      rbm_abs_addr(get_rbm_addr(tail)),
       std::move(delta_handler),
-      segment_seq_t(NULL_SEG_SEQ),
-      [this, FNAME](auto &is_rolled, auto &cursor_addr, auto &d_handler, auto &expected_seq) {
-      return crimson::repeat(
-	[this, &is_rolled, &cursor_addr, &d_handler, &expected_seq, FNAME]() mutable
-	-> replay_ertr::future<seastar::stop_iteration> {
-	paddr_t record_paddr = convert_abs_addr_to_paddr(
-	  cursor_addr,
-	  get_device_id());
-	return read_record(record_paddr, expected_seq
-	).safe_then([this, &is_rolled, &cursor_addr, &d_handler, &expected_seq, FNAME](auto ret)
-	    -> replay_ertr::future<seastar::stop_iteration> {
-	  if (!ret.has_value()) {
-	    if (expected_seq == NULL_SEG_SEQ || is_rolled) {
-	      DEBUG("no more records, stop replaying");
-	      return replay_ertr::make_ready_future<
-		seastar::stop_iteration>(seastar::stop_iteration::yes);
-	    } else {
-	      cursor_addr = get_records_start();
-	      ++expected_seq;
-	      is_rolled = true;
-	      return replay_ertr::make_ready_future<
-		seastar::stop_iteration>(seastar::stop_iteration::no);
+      std::map<paddr_t, journal_seq_t>(),
+      [this](auto &d_handler, auto &map) {
+      auto add_invalid_records = [&map](
```

Comment:
Probably better named `build_paddr_seq_map` according to its goal.

Reply:
Done.